sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/constraints/Check.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connector.catalog.constraints;
+
+import java.util.Map;
+import java.util.Objects;
+
+import org.apache.spark.SparkIllegalArgumentException;
+import org.apache.spark.sql.connector.expressions.filter.Predicate;
+
+/**
+ * A CHECK constraint.
+ * <p>
+ * A CHECK constraint defines a condition each row in a table must satisfy. Connectors can define
+ * such constraints either in SQL (Spark SQL dialect) or using a {@link Predicate predicate} if the
+ * condition can be expressed using a supported expression. A CHECK constraint can reference one or
+ * more columns. Such constraint is considered violated if its condition evaluates to {@code FALSE},
+ * but not {@code NULL}. The search condition must be deterministic and cannot contain subqueries
+ * and certain functions like aggregates or UDFs.
+ *
+ * @since 4.1.0
+ */
+public class Check extends BaseConstraint {
```

Comment:
Shall we override the `enforced()`/`rely()`/`validationStatus()` method or variable? It will help developers to understand them.
Otherwise, the default values will only exist in docs and spark internal code.

Reply:
I am not sure it is a good idea. Those methods existing in the parent interface. The method will show up in Javadoc and IDE will suggest them as well. I am worried about all code duplication because of this.

Reply:
As a compromise, we can add more Javadoc at the top of this class. What do you think?

Reply:
Adding more java doc seems ok.
If we change the default value in these classes, it can make the internal implementation simpler too. For example, we don't need to have the default `ConstraintCharacteristic` for each internal parsed constraints https://github.com/gengliangwang/spark/pull/13/files#diff-03507453aabc732a7e3efadc81cd840e436a392b1c62d5aa50647266bc3a9199R38

Reply:
Added more Javadoc. 

> If we change the default value in these classes, it can make the internal implementation simpler too. For example, we don't need to have the default ConstraintCharacteristic for each internal parsed constraints

Let's re-evaluate this in the implementation PR. I think we can either remove the common builder for constraints or handle this in the parser.